internal/testcontainersdocker/docker_host.go

```diff
@@ -30,32 +42,81 @@ func DefaultGatewayIP() (string, error) {
 }
 
 // Extracts the docker host from the context, or returns the default value
-func ExtractDockerHost(ctx context.Context) (dockerHostPath string) {
-	if dockerHostPath = os.Getenv("TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE"); dockerHostPath != "" {
-		return dockerHostPath
+func ExtractDockerHost(ctx context.Context) string {
```

Comment:
@HofmeisterAn this is the entrypoint to the docker_host discovery. At this moment it's not called when getting the docker client as I wanted to make it sure I was keeping the old behaviour first